### DISCUSSION TITLE
Introduce a 'series of post' feature

### DISCUSSION THREAD

**Discussion by ZzMzaw** [welpo/tabi#256] [Ideas]
Hello,

I was wondering if it could be interesting to have a 'series of post' feature.
Each post of the serie would have a kind a block at the beginning indicating that the post is part of a serie, a link to the serie page (providing an introduction and the ordered list of posts part of the serie), a link to the previous post and a link to the next one.
Posts of a serie would also be part of all the other posts as usual (i.e. ordered by date and so on) but would also be accessible through the serie with the right order.

I saw you already did something similar in your personal blog and the result is really close to what I was looking for.
Nevertheless, I was wondering if using tags would be easy enough to maintain.
The drawback I see is that the serie introduction is not unique (you need to update it in each post of the serie), it is hard to know which post is part of a serie just looking at the folder tree and I am not sure if the order will remain in case of post update (but I would assume yes).

Even if I am not sure if it would be feasible (kind of nested posts), I had the following structure in mind:
- folder `content`
  - folder `blog`
    - folder `post A`
      - file `index.md`
    - folder `serie of posts B`
      - file `index.md` with the description of the serie
      - folder `post 1` for the first post of the serie
        - file `index.md`
      - folder `post 2` for the second post of the serie
        - file `index.md`
      - ...
    - folder `post C`
      - file `index.md`
    - ...

Order of the posts of the serie could rely on the alphabetical order in the `serie B` folder.

Anyway, even if I won't have much time to work on that for now, I would be happy to know what you think about it.

Thanks again for all the great work with tabi.

**Comment by welpo** [comment]
Hello!

I definitely considered spending some time coding this when I started the series in my blog. It's hard to balance "working on my blog" and "working on the infrastructure of my blog".

For my use-case, I don't mind copypasting the intro and modifying it a bit. However, I do think it would make sense to add this feature; other users might have many series. This would save them a lot of work.

Some thoughts:
- The nested structure you propose both makes sense and should work. Haven't tested it, though.
- I assume the folder `serie of posts B` would need to be a section. Thus, we'd have `_index.md` inside.
- In `_index.md`, we could place an intro to the series in the `[extra]` section. Perhaps with placeholders like `$POST_NUMBER` and `$TOTAL_POSTS` that would get replaced in the Tera template. This variable would be added as a paragraph on each post.
- The order of the posts should be determined by the section configuration (`sort_by`). I'd personally prefer by date, but this would be up to the user.
- The next/previous post navigation added in  #246 should automatically list the next/previous article **in the series**.

The next steps would be:
1. Confirm nested sections work as expected in terms of rendering posts.
2. I believe the `section.html` would need to be modified to render not just all posts, but also sections nested inside it.

I don't have much time to code this myself, but I'm happy to test changes and provide guidance where needed!

**Reply by ZzMzaw** [reply]
Thanks for your feedback.

I'll try to work on it when I have some time in the coming weeks/months.

I keep you posted as soon as I have something demonstrable.

**Comment by ZzMzaw** [comment]
Hello,

After such a long time, I just proposed an implementation in PR #360 .

The PR is still draft in order to agree on it. I'll refine it once agreed to reach tabi's quality standards with all required documentation and examples.

Thanks for your help and support.

**Comment by ZzMzaw** [comment]
The feature has now been implemented and merged into the main branch of tabi.
I'll close this discussion as resolved.
Feel free to reopen it if you think it is needed.
